web/client/components/geostory/media/Video.jsx

```diff
@@ -200,32 +230,47 @@ const VideoMedia = ({
         onPlay(newPlaying);
     };
 
+    // reset player after switching from view to edit mode
+    useEffect(() => {
+        if (mode === Modes.EDIT) {
+            handleOnPlay(false);
+            setStarted(false);
+        }
+    }, [ mode ]);
+
+    // enable autoplay only the first time the video is in view for fit undefined or contained
+    // while fit cover should always trigger autoplay when in view
     useEffect(() => {
-        if (inView && autoplay && !started) {
+        if (mode === Modes.VIEW
+        && isVisible
+        && (fit === 'cover' || autoplay && !started)) {
```

Comment:
I changed fit === 'cover' in isCover in the Video component as you suggested but not this one (VideoMedia) because here fit === 'cover' is used only once in the `useEffect` and fit it'a a dependency

Reply:
ok